contentcuration/contentcuration/static/js/edit_channel/utils/drag_drop.js

```diff
@@ -149,11 +153,31 @@ function addButtonDragDrop(element, dropCallback){
         		selected_items.push(current_node);
         	}
         	selected_items.add(appended_items.models, {at: selected_items.length});
-        	dropCallback(selected_items).then(function(){
-        		$(ui.draggable.context).remove();
-        		$(".content-list").sortable( "enable" );
-        		$(".content-list").sortable( "refresh" );
-        	});
+
+        	// Check for related content and display warning if found
+        	if(selected_items.has_related_content()) {
+        		dialog.dialog(stringHelper.get_translation(messages, "related_content"), stringHelper.get_translation(messages, "related_content_warning", selected_items.length), {
+	                [stringHelper.get_translation(messages, "cancel")]:function(){
+	                	$(".content-list").sortable("cancel" );
+		        		$(".content-list").sortable( "enable" );
+		        		$(".content-list").sortable( "refresh" );
```

Comment:
indentation could a little cleanup here!  this area reminds me of that clipboard meeting